examples/client/tui/commands.nim

```diff
@@ -780,11 +778,151 @@ proc command*(self: Tui, command: CallRpc) {.async, gcsafe,
     if command.rpcMethod == "":
       self.wprintFormatError(getTime().toUnix,
         "method cannot be blank, please provide an input.")
-    elif command.params.kind == JNull:
+      return
+    
+    var params = newJArray()
+    if command.params != "":
+      try:
+        params = command.params.parseJson()
+      except:
+        self.wprintFormatError(getTime().toUnix,
+          "params must be a valid JSON")
+        return
+
+    asyncSpawn self.client.callRpc(command.rpcMethod, params)
+  except:
+    self.wprintFormatError(getTime().toUnix, "invalid arguments.")
+
+# SendTransaction --------------------------------------------------------------
+
+proc help*(T: type SendTransaction): HelpText =
+  let command = "sendtransaction"
+  HelpText(command: command, aliases: aliased[command], parameters: @[
+    CommandParameter(name: "from", description: "The address for the sending account"),
+    CommandParameter(name: "to", description: "The destination address of the message, left empty for a contract-creation transaction."),
+    CommandParameter(name: "value", description: "The value transferred for the transaction in wei"),
+    CommandParameter(name: "maxPriorityFee", description: "Max priority fee in wei"),
+    CommandParameter(name: "maxFee", description: "Max fee in wei"),
+    CommandParameter(name: "gasLimit", description: "The amount of gas to use for the transaction (unused gas is refunded)."),
+    CommandParameter(name: "data", description: "ABI byte string containing the data of the function call on a contract, or contract-creation initialisation code."),
+    CommandParameter(name: "nonce", description: "Integer of the nonce"),
+    CommandParameter(name: "password", description: "Account password")
+    ], description: "Sends an EIP1559 transaction")
+
+proc new*(T: type SendTransaction, args: varargs[string]): T =
+  T(fromAddress: args[0],
+    toAddress: args[1],
+    value: args[2],
+    maxPriorityFee: args[3],
+    maxFee: args[4],
+    gasLimit: args[5],
+    payload: args[6],
+    nonce: args[7],
+    password: args[8])
+
+proc split*(T: type SendTransaction, argsRaw: string): seq[string] =
+  let args = argsRaw.split(" ")
+  let
```

Comment:
What if instead of all the `if/else`, an array.of string with length 9 is initialized? Then you could do `for a in args: ...` and overwrite the empty strings, and return the array converted to a seq.